lib/runtime/src/utils/tasks/tracker.rs

```diff
@@ -0,0 +1,3551 @@
+            &format!("{}_tasks_rejected_total", component_name),
+            "Total number of rejected tasks",
+            &[],
+        )?;
+
+        let active_gauge = registry.create_intgauge(
+            &format!("{}_tasks_active", component_name),
+            "Current number of active tasks",
+            &[],
+        )?;
+
+        let queued_gauge = registry.create_intgauge(
+            &format!("{}_tasks_queued", component_name),
+            "Current number of tasks queued in scheduler",
+            &[],
+        )?;
+
+        Ok(Self {
+            local_metrics: TaskMetrics::new(),
+            prometheus_issued: issued_counter,
+            prometheus_success: success_counter,
+            prometheus_cancelled: cancelled_counter,
+            prometheus_failed: failed_counter,
+            prometheus_rejected: rejected_counter,
+            prometheus_active: active_gauge,
+            prometheus_queued: queued_gauge,
+        })
+    }
+}
+
+impl RootTaskMetrics {
+    /// Update the queued gauge based on current metrics
+    fn update_queued_gauge(&self) {
+        let queued = self.local_metrics.queued() as i64;
+        self.prometheus_queued.set(queued);
+    }
+}
+
+impl HierarchicalTaskMetrics for RootTaskMetrics {
+    fn increment_issued(&self) {
+        self.local_metrics.increment_issued();
+        self.prometheus_issued.inc();
+        self.update_queued_gauge();
+    }
+
+    fn increment_success(&self) {
+        self.local_metrics.increment_success();
+        self.prometheus_success.inc();
+        self.update_queued_gauge();
+    }
+
+    fn increment_cancelled(&self) {
+        self.local_metrics.increment_cancelled();
+        self.prometheus_cancelled.inc();
+        self.update_queued_gauge();
+    }
+
+    fn increment_failed(&self) {
+        self.local_metrics.increment_failed();
+        self.prometheus_failed.inc();
+        self.update_queued_gauge();
+    }
+
+    fn increment_rejected(&self) {
+        self.local_metrics.increment_rejected();
+        self.prometheus_rejected.inc();
+        self.update_queued_gauge();
+    }
+
+    fn increment_active(&self) {
+        self.local_metrics.increment_active();
+        self.prometheus_active.inc();
+        self.update_queued_gauge();
+    }
+
+    fn decrement_active(&self) {
+        self.local_metrics.decrement_active();
+        self.prometheus_active.dec();
+        self.update_queued_gauge();
+    }
+
+    fn issued(&self) -> u64 {
+        self.local_metrics.issued()
+    }
+
+    fn success(&self) -> u64 {
+        self.local_metrics.success()
+    }
+
+    fn cancelled(&self) -> u64 {
+        self.local_metrics.cancelled()
+    }
+
+    fn failed(&self) -> u64 {
+        self.local_metrics.failed()
+    }
+
+    fn rejected(&self) -> u64 {
+        self.local_metrics.rejected()
+    }
+
+    fn active(&self) -> u64 {
+        self.local_metrics.active()
+    }
+
+    fn prometheus_metrics(&self) -> Option<Result<String>> {
```

Comment:
would it make sense to leave this as a commented todo?

Reply:
this was a placeholder, I should remove it.  because we are creating the metrics objects thru the newly created dynamo metrics registry, the metrics object are already registered for us.